app/controllers/pet_applications_controller.rb

```diff
@@ -0,0 +1,6 @@
+class PetApplicationsController < ApplicationController
+  def create
+    PetApplication.create!(pet: Pet.find(params[:pet]), application: Application.find(params[:id]))
+    redirect_to "/applications/#{params[:id]}"
```

Comment:
Interesting logic, I was thinking how the controller would've handled this action.